### PR TITLE
Ada learning 0003/refactoring/speadsheat+gmail notification

### PR DESCRIPTION
**Pull request by a-da**

### CODE REVIEW

File: google.com/src/ada_google_com/sheets.py [hunk 1]
```diff
@@ -1,133 +1,239 @@
...
+            ) from index_error
+
+        regexp = re.compile(
+            r'Notify if when: more than (\d+) days',
+            re.IGNORECASE
+        )
+        match = regexp.search(notify_after_n_days_raw_value)
+        if not match:
+            print(f'Can not match {regexp.pattern}')
+            sys.exit(1)
```
Comment (galij899) [inline google.com/src/ada_google_com/sheets.py]: Why using such rough syntax? Can't a proper exception be raised there?

File: google.com/src/ada_google_com/sheets.py [hunk 2]
```diff
@@ -1,133 +1,239 @@
...
         date = datetime.strptime(last_date_checked_in, '%Y.%m.%d')
-        if (now - date) > timedelta(days=notify_after_n_days):
-            need_notification = True
-            output.append((
-                'you have to log into to refresh account, '
-                f'you could loose that account .... {repr(name)!r}, {user!r}, {email!r}')
+        if (now - date) > timedelta(days=self.foot.notify_after_n_days):
+            return (
+                'You have to log into to refresh account, '
+                f'you could loose that account .... {repr(name)!r}, {user!r}, {email!r}'
```
Comment (galij899) [inline google.com/src/ada_google_com/sheets.py]: loose -> lose

Comment (galij899) [reply google.com/src/ada_google_com/sheets.py]: Also repr(name) is unnecessary if **!r** flag is used in f-string

File: google.com/src/ada_google_com/sheets.py (hunk 1 above)
Comment (a-da) [reply google.com/src/ada_google_com/sheets.py]: fixed

File: google.com/src/ada_google_com/sheets.py (hunk 2 above)
Comment (a-da) [reply google.com/src/ada_google_com/sheets.py]: fixed.

**Comment by a-da** [comment]
thanks @galij899

**Pull request merged**